server/internal/infrastructure/gcp/config.go

```diff
@@ -0,0 +1,21 @@
+package gcp
+
+import "fmt"
+
+type Config struct {
+	cloudTasks *CloudTasksConfig
+}
+
+type CloudTasksConfig struct {
+	GCPProject    string
+	GCPRegion     string
+	QueueName     string
+	SubscriberURL string
+}
+
+func (c *CloudTasksConfig) buildQueueUrl() (string, error) {
+	if c.GCPProject == "" || c.GCPRegion == "" || c.QueueName == "" {
+		return "", ErrMissignConfig
+	}
+	return fmt.Sprintf("projects/%s/locations/%s/queues/%s", c.GCPProject, c.GCPRegion, c.QueueName), nil
```

Comment:
Sorry, I couldn't understand well what you mean. 🤔 

To make the queue URL which consist with `project`, `region` and `queue name`, we must have those at least before the `CreateTask` function is called by cloudtasks client.

One choice we can have is call `buildQueueURL` function when `CreateTask` function is called ( not when Client is initialized ) but if we choose the way, if necessary config isn't passed the function will return the error. I believe it's better to return error when it's initialized rather than when is executed.

Reply:
I've tested and also deeply had a look SDK code base but I assume it doesn't take care of these configs which means we need to pass these configs for cloudtasks client. 

The steps I did
1. run `gcloud auth application-default login` and get the credential for local development
2. run `Run` method but SDK didn't build queue URL by detecting default GCP project and region.